### PR TITLE
[Testing] Reduce image build flakiness by share and retry cloudbuild jobs

### PR DESCRIPTION
**Pull request by Bobgy**
Part of https://github.com/kubeflow/pipelines/issues/3256

Changes:
* All 3 presubmit tests will share one cloudbuild job for image building
* Add retry at most twice if the first build fails
* (not mentioned in title, but it's a simple change) Added retry to workload identity setup

Flakiness rate should be greatly reduced.

Previously, we had 3 presubmit tests and 1 cloudbuild job for each presubmit test.
If any one of them fails, developer need to retry at least one test.
With this PR, all 3 presubmit tests share one cloudbuild job and they can retry it 2 times (total 3 attempts).

e.g. current data for image building is averagely 10% jobs fail randomly.

Previous flakiness rate was 1-(1-10%)^3 = 27.1%.
With this PR, it will be 10%^3 = 0.1%.

### CODE REVIEW

File: test/build-images.sh [hunk 1]
```diff
@@ -18,40 +18,51 @@ set -ex
...
   echo "docker images for api-server, frontend, scheduledworkflow, \
     persistenceagent, viewer-crd-controller, inverse-proxy-agent, metadata-writer, cache-server, \
     cache-deployer and visualization-server are already built in ${GCR_IMAGE_BASE_DIR}."
 else
-  echo "submitting cloud build to build docker images for commit ${COMMIT_SHA}..."
-  IMAGES_BUILDING=true
-  CLOUD_BUILD_COMMON_ARGS=(. --async --format='value(id)' --substitutions=_GCR_BASE=${GCR_IMAGE_BASE_DIR})
-  # Split into two tasks because api_server builds slowly, use a separate task
-  # to make it faster.
-  BUILD_ID_API_SERVER=$(gcloud builds submit ${CLOUD_BUILD_COMMON_ARGS[@]} \
```
Comment (Bobgy) [inline test/build-images.sh:47]: It's too complex to keep track of both cloudbuild tasks, I merged both tasks to one.

Comment (Bobgy) [reply test/build-images.sh:47]: I figured out how to deal with two builds.

**Comment by Bobgy** [comment]
/test kubeflow-pipeline-sample-test
/test kubeflow-pipeline-upgrade-test
/test kubeflow-pipeline-e2e-test
verify again

**Comment by Bobgy** [comment]
@numerology @Ark-kun @rmgogogo I've retried three times without one failure.
Can you please review?

File: test/build-images.sh [hunk 2]
```diff
@@ -18,40 +18,77 @@ set -ex
...
 then
-  echo "docker images for api-server, frontend, scheduledworkflow, \
+  echo "docker images for frontend, scheduledworkflow, \
     persistenceagent, viewer-crd-controller, inverse-proxy-agent, metadata-writer, cache-server, \
     cache-deployer and visualization-server are already built in ${GCR_IMAGE_BASE_DIR}."
 else
-  echo "submitting cloud build to build docker images for commit ${COMMIT_SHA}..."
   IMAGES_BUILDING=true
-  CLOUD_BUILD_COMMON_ARGS=(. --async --format='value(id)' --substitutions=_GCR_BASE=${GCR_IMAGE_BASE_DIR})
-  # Split into two tasks because api_server builds slowly, use a separate task
```
Comment (Ark-kun) [inline test/build-images.sh:45]: Maybe, keep this comment?

**Comment by Ark-kun** [comment]
Thank you a lot for fixing the flakiness!
/lgtm

**Comment by numerology** [comment]
/lgtm

Thanks @Bobgy !

File: test/build-images.sh (hunk 2 above)
Comment (Bobgy) [reply test/build-images.sh:45]: good point!

**Pull request merged**